### PR TITLE
(feat): remove ability for non-marketplace owner to switch shops

### PR DESCRIPTION
**Pull request by kieckhafer**
This update changes the shop selector dropdown to only allow marketplace owners to switch shops. If you are a single shop owner, the selector will no longer be displayed, and you will no longer be allowed to switch active shops.

To test:
- Login as Reaction admin
- Invite new shop owner
- Make new shop active
- See you can switch back and forth between shops as Marketplace owner
- Make sure the last shop you land on is the original Marketplace shop (this will test to make sure the setting isn't saved in Browser storage)

THEN:

- Login as new shop
- See shop selector is not displayed, AND you are able to make all admin actions on your shop (i.e. the active shop is your shop, not the marketplace shop)

### CODE REVIEW

**Comment by kieckhafer** [comment]
@spencern this works as-is in this PR, however, please see comment on the original ticket (https://github.com/reactioncommerce/reaction/issues/3184#issuecomment-349769921) for perhaps an update we can make to make it a little better.

**Comment by spencern** [comment]
@kieckhafer left some comments over there on #3184 - we can pair on this if you like.

What I think we need to make 100% certain of in this PR is that on login a user's activeShop is set correctly, and on logout it's set back to the primary shop correctly.

File: client/modules/core/main.js [hunk 1]
```diff
@@ -298,6 +298,24 @@ export default {
     return typeof hasPermissions !== "undefined";
   },
 
+  getShopsForUser(roles, userId = Meteor.userId()) {
+    const shops = [];
+    const user = Meteor.user(userId);
+
+    Object.keys(user.roles).find((shopId) => {
```
[the remark this answers is not in the document]
Comment (kieckhafer) [reply client/modules/core/main.js]: Ok. I know we discussed `reduce` earlier, but once coding, I saw this `find` inside the `hasDashboardAccessForAnyShop`, and figured I'd keep the code as similar as I could to other code we'd written.

**Comment by kieckhafer** [comment]
@spencern updated to use reduce.

One last thing I was noticing... Marketplace owners will always have the switcher, even if there is only one shop, because of the `__global_roles__` group that's attached to them.

Should we account for this, or is this OK to always show for the main owner?

**Comment by spencern** [comment]
@kieckhafer good catch, we should probably exclude the `__global_roles__` group from this calculation

**Comment by kieckhafer** [comment]
@spencern updated to remove `__global_roles__`

**Comment by spencern** [comment]
@kieckhafer yeah, that would be better because it would not loop through all roles unnecessarily for the `__global_roles__` group.

**Comment by kieckhafer** [comment]
@spencern cool, updated.

**Pull request merged**